File: Atcoder_#328/javascript/b_1111.js
function main(input) {
  let lines = input.trim().split('\n');
  let n = parseInt(lines[0]);
  let days = lines[1].split(' ').map(Number);

  let count = 0;

  for (let i = 1; i <= n; i++) {
    for (let j = 1; j <= days[i - 1]; j++) {
      if (new Set(String(i)).size === 1 && new Set(String(j)).size === 1) {
        if (String(i)[0] === String(j)[0]) {
          count++;
        }
      }
    }
  }

  console.log(count);
}

main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: Atcoder_#328/ruby/b_1111.rb
n = gets.to_i
d = gets.split.map(&:to_i)

count = 0

(1..n).each do |i|
  (1..days[i-1]).each do |j|
    if i.to_s.chars.uniq.size == 1 && j.to_s.chars.uniq.size == 1
      count += 1 if i.to_s[0] == j.to_s[0]
    end
  end
end

puts count

